res/annotation-grammar.pp

```diff
@@ -0,0 +1,86 @@
+%pragma lexer.unicode 1
+
+%skip   space              [\x20\x09\x0a\x0d]+
+%token  doc_               /\*\*                      -> docblock
+
+%skip   docblock:space     [\x20\x09\x0a\x0d]+
+%skip   docblock:star      \*(?!/)
+%token  docblock:_doc      \*/                         -> default
+%token  docblock:at        @(?!\s)                     -> annot
```

Comment:
Maybe update the token to this:

```
%token docblock:at (?<!\b)@(?=\b)
```

Testing:

```
(1)  * @bcdef
(2)  * @ bcdef
(3)  * @author <hello@world>
(4)  *@xyz
```

Case 1, the `@` is selected, as expected.
Case 2, the `@` is _not_ selected, as expected.
Case 3, the `@` in `@author` is selected, and the `@` in the email address is _not_ selected, as expected.
Case 4, the `@` is selected, as expected.

See regexr.com/42ktg

Thoughts?

Reply:
I just though that `hello-@world` is valid email address too, so the token might be updated to:

```diff
- %token docblock:at (?<!\b)@(?=\b)
+ %token docblock:at (?<=[\h\*])@(?=\b)
```

Use the same https://regexr.com/42ktg link, with a new test case:

```
(5)  *@xyz <hello-@world>
```

Case 5, the `@` in `@xyz` is selected, and the `@` in the email address is _not_ selected, as expected.

Thoughts?